### PR TITLE
added Feature keep on display

### PR DESCRIPTION
**Pull request by jonycoo**
I've already Implemented it,
if you want Include it in the next release
Resolve Feature of Issue #54
please reply if I need to change anything before merge.
Thanks

### CODE REVIEW

File: app/src/main/java/org/kaqui/testactivities/TestActivity.kt [hunk 1]
```diff
@@ -76,6 +76,9 @@ class TestActivity : BaseActivity(), TestFragmentHolder, CoroutineScope {
 
         job = Job()
 
+        if(applicationContext.defaultSharedPreferences.getBoolean("keep_on",false))
+            window.addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON)
```
Comment (blastrock) [inline app/src/main/java/org/kaqui/testactivities/TestActivity.kt:80]: Please format the files. It's Alt+Shift+L on Android Studio.

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -130,6 +130,10 @@
     <string name="show_answers">Show answers</string>
     <string name="failed_to_load_font">Failed to load custom font: %s</string>
     <string name="failed_to_load_resource">Failed to load resource: %s</string>
+    <string name="keep_on_sum">Screen keeps lit in Study</string>
```
Comment (blastrock) [inline app/src/main/res/values/strings.xml:133]: Can you rename the string to "keep_screen_on_summary"? Also, I'd write "Keep screen lit during tests".

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -130,6 +130,10 @@
...
+    <string name="screen_keep_on">Keep on</string>
```
Comment (blastrock) [inline app/src/main/res/values/strings.xml:134]: Can you rename this string to keep_screen_on? And write "Keep screen on" as the option name.

**Review by blastrock — CHANGES_REQUESTED** [review]
Thanks a lot for the contribution!

I have left you a few comments. Also, if you want to, you can add your name in the AUTHORS file with description like "keep screen on option".

**Review by jonycoo — COMMENTED** [review]
changes committed in d749b5fb065c0fe78285e37b5cc9e3e4fd257ce7

**Comment by musotec** [comment]
Apologies if I state anything wrong about the architecture, I only now have taken a cursory look at the structure of this app.

I think it would also be preferable to have keep awake on only when the user is in a TestFragment. This way a user either viewing stats or leaving any of the quiz selection windows open does not keep the screen on and draining battery.

Consider adding the removal of the keep awake window flag on the fragment transition (https://github.com/blastrock/kakugo/blob/47e28993165268f8da4461d873eee71f1369b0a1/app/src/main/java/org/kaqui/testactivities/TestActivity.kt#L307)

Window flags can be cleared using:
`window.clearFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON)`

This is recommended in this official documentation: https://developer.android.com/training/scheduling/wakelock#screen

**Comment by blastrock** [comment]
> I think it would also be preferable to have keep awake on only when the user is in a TestFragment. This way a user either viewing stats or leaving any of the quiz selection windows open does not keep the screen on and draining battery.

This PR puts the flag on the TestActivity, which always contain a TestFragment. I don't think putting it there would change anything.

From what I understand from the documentation, it is fine to put it in the activity, as @jonycoo did. Android will disable the feature when you switch app, or when you go back I guess... It will be easy to test.

If it doesn't work that way, we can set the flag and clear it in onStart/onStop.

**Review by blastrock — APPROVED** [review]
It does work, the screen is kept on only when the test activity is open.

Thanks for your contribution!

**Pull request merged**